go/geth_adapter/adapter.go

```diff
@@ -85,7 +85,11 @@ func (a *gethInterpreterAdapter) Run(contract *geth.Contract, input []byte, read
 	a.evm.SetDepth(a.evm.GetDepth() + 1)
 	defer func() { a.evm.SetDepth(a.evm.GetDepth() - 1) }()
 
-	revision := convertRevision(a.evm.ChainConfig(), a.evm.Context.BlockNumber, a.evm.Context.Time)
+	rules := a.evm.ChainConfig().Rules(a.evm.Context.BlockNumber, a.evm.Context.Random != nil, a.evm.Context.Time)
+	revision, err := convertRevision(rules)
+	if err != nil {
+		return nil, fmt.Errorf("unsupported revision: %v", err)
```

Comment:
is this error covered by any test ?

Reply:
There is a test which checks that the `convertRevision` function returns an error which is the important part. All the errors in the run function should be checked in unit tests but this will be a separate PR.